GUI.py

```
@@ -0,0 +1,747 @@
+
+class FilterThread(QThread):
+    finished = pyqtSignal(str)
+    error = pyqtSignal(str)
+
+    def __init__(self, command):
+        super().__init__()
+        self.command = command
+
+    def run(self):
+        try:
+            result = subprocess.run(self.command, shell=True, capture_output=True, text=True)
```

Ruff (0.12.2)

`subprocess` call with `shell=True` identified, security issue

(S602)